test/e2e/framework/skip.go

```diff
@@ -136,18 +150,50 @@ func SkipIfNodeOSDistroIs(unsupportedNodeOsDistros ...string) {
 	}
 }
 
+// SkipUnlessPrometheusMonitoringIsEnabled skips if the prometheus monitoring is not enabled.
+func SkipUnlessPrometheusMonitoringIsEnabled(supportedMonitoring ...string) {
+	if !TestContext.EnablePrometheusMonitoring {
+		skipInternalf(1, "Skipped because prometheus monitoring is not enabled")
+	}
+}
+
+// SkipUnlessServerVersionGTE skips if the server version is less than v.
+func SkipUnlessServerVersionGTE(v *utilversion.Version, c discovery.ServerVersionInterface) {
+	gte, err := ServerVersionGTE(v, c)
+	if err != nil {
+		Failf("Failed to get server version: %v", err)
+	}
+	if !gte {
+		skipInternalf(1, "Not supported for server versions before %q", v)
+	}
+}
+
+// SkipUnlessSSHKeyPresent skips if no SSH key is found.
+func SkipUnlessSSHKeyPresent() {
+	if _, err := e2essh.GetSigner(TestContext.Provider); err != nil {
+		skipInternalf(1, "No SSH Key for provider %s: '%v'", TestContext.Provider, err)
+	}
+}
+
 // SkipUnlessTaintBasedEvictionsEnabled skips if the TaintBasedEvictions is not enabled.
 func SkipUnlessTaintBasedEvictionsEnabled() {
 	if !utilfeature.DefaultFeatureGate.Enabled(features.TaintBasedEvictions) {
 		skipInternalf(1, "Only supported when %v feature is enabled", features.TaintBasedEvictions)
 	}
 }
 
-// SkipIfContainerRuntimeIs skips if the container runtime is included in the runtimes.
-func SkipIfContainerRuntimeIs(runtimes ...string) {
-	for _, containerRuntime := range runtimes {
-		if containerRuntime == TestContext.ContainerRuntime {
-			skipInternalf(1, "Not supported under container runtime %s", containerRuntime)
-		}
+// ServerVersionGTE returns true if v is greater than or equal to the server
+// version.
+//
+// TODO(18726): This should be incorporated into client.VersionInterface.
+func ServerVersionGTE(v *utilversion.Version, c discovery.ServerVersionInterface) (bool, error) {
```

Comment:
ServerVresionGTE is called only from SkipUnlessServerVersionGTE which is moved to skip.go together.
Can we make the function local as `serverVersionGTE`?

Reply:
ha :sweat_smile: forgot to check this. thank you for the catch!
fixed